File: SrdDatabase/ClientApp/src/models/congregation.ts
import { PagedParameters, PagedResults, Population } from './shared';
import { EventResults } from './event'
import { PaymentResults } from './payment';
import { QuotaResults } from './quota';

export interface Congregation {
    id?: number;
    name?: string;
    parishId?: number;
    parish?: string;
    archdeaconryId?: number;
    archdeaconry?: string;
    numberOfChristians?: number;
    quota?: number;
    balance?: number;
}

export interface CongregationDetails {
    congregation: Congregation;
    population: Population;
    eventResults: EventResults;
    paymentResults: PaymentResults;
    quotaResults: QuotaResults;
}

export interface CongregationParameters extends PagedParameters {
    name?: string;
    parishId?: number;
    archdeaconryId?: number;
}

export interface CongregationResults extends PagedResults {
    congregations: Congregation[];
}

File: SrdDatabase/Models/Congregations/Congregation.cs
namespace SrdDatabase.Models.Congregations
{
    public class Congregation
    {
        public int Id { get; }

        public string Name { get; }

        public int ParishId { get; }

        public string Parish { get; }

        public int ArchdeaconryId { get; }

        public string Archdeaconry { get; }

        public int? NumberOfChristians { get; }

        public int Quota { get; }

        public int Balance { get; }

        public Congregation(
            int id,
            string name,
            int parishId,
            string parish,
            int archdeaconryId,
            string archdeaconry,
            int? numberOfChristians,
            int quota,
            int balance
            )
        {
            Id = id;
            Name = name;
            ParishId = parishId;
            Parish = parish;
            ArchdeaconryId = archdeaconryId;
            Archdeaconry = archdeaconry;
            NumberOfChristians = numberOfChristians;
            Quota = quota;
            Balance = balance;
        }

        // for Dapper
        public Congregation()
        {
        }
    }
}
